File: modules/social-video-export/src/SocialVideoExport.types.ts
// Parametri per composeSocialVideo. Vedi android/.../SocialVideoExportTypes.kt
// per il record nativo corrispondente (stessi nomi di campo).
export type ComposeSocialVideoOptions = {
  /** uri (file://) del video o della gif da usare come sfondo animato */
  sourceUri: string;
  /** uri (file://) del PNG della card già catturato lato JS (con alpha) */
  cardUri: string;
  /** uri (file://) di destinazione del mp4 risultante */
  outputUri: string;
  canvasWidth: number;
  canvasHeight: number;
  /** geometria della card sul canvas, già risolta lato JS (getCardPlacement) */
  cardX: number;
  cardY: number;
  cardWidth: number;
  cardHeight: number;
  /** opacità del velo scuro, 0-255 (~89 = 0.35 di opacità) */
  veilAlpha?: number;
  /** durata massima esportata dalla sorgente, in millisecondi (default 15000) */
  maxDurationMs?: number;
  /** frame al secondo dell'output (default 12) */
  fps?: number;
};

File: modules/social-video-export/android/src/main/java/expo/modules/socialvideoexport/SocialVideoExportTypes.kt
package expo.modules.socialvideoexport

import expo.modules.kotlin.exception.CodedException
import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

/**
 * Parametri per composeSocialVideo. La geometria della card (cardX/Y/W/H) arriva
 * già risolta dal lato JS (vedi src/services/socialCardCompose.js →
 * getCardPlacement), così la logica dei preset/safe-zone resta in un unico posto:
 * qui il nativo fa solo "pixel pushing" (decodifica sorgente, compone, incoda).
 */
class ComposeOptions : Record {
  @Field val sourceUri: String = ""
  @Field val cardUri: String = ""
  @Field val outputUri: String = ""
  @Field val canvasWidth: Int = 1080
  @Field val canvasHeight: Int = 1920
  @Field val cardX: Double = 0.0
  @Field val cardY: Double = 0.0
  @Field val cardWidth: Double = 0.0
  @Field val cardHeight: Double = 0.0
  @Field val veilAlpha: Int = 89 // ~0.35 * 255, coerente con VEIL_COLOR in socialCardCompose.js
  @Field val maxDurationMs: Double = 15000.0
  @Field val fps: Int = 12
}

internal class SocialVideoExportException(message: String, cause: Throwable? = null) :
  CodedException(message, cause)
